src/library/base/R/dataframe.R

```diff
@@ -598,14 +598,14 @@ data.frame <-
 ###  These are a little less general than S
 
 `[.data.frame` <-
-    function(x, i, j, drop = if(missing(i)) TRUE else length(cols) == 1)
+    function(x, i, j, drop = if(missing(i)) TRUE else length(cols) == 1, pmatch.rows = TRUE)
```

Comment:
Should there be an option like `warnPartialMatch{Args,Attr,Dollar}` to throw a warning if code is relying on the default behavior?

Reply:
Just saw your comment later in the thread implying you won't pursue this as a patch :) resolving